### PR TITLE
engine: Adds new helper functions for default exchange deployment. 

### PR DESCRIPTION
**Pull request by shazbert**
# PR Description

De-couples switch case and adds helper function to generate a new value defaulted exchange which can be used straight away for external usage. 

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration and
also consider improving test coverage whilst working on a certain feature or package.

- [x] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gloriousCode — CHANGES_REQUESTED** [review]

**Comment by shazbert** [comment]
#1245

File: engine/helpers.go [hunk 1]
```diff
@@ -963,3 +990,86 @@ func genCert(targetDir string) error {
...
+		return new(zb.ZB), nil
+	default:
+		return nil, fmt.Errorf("'%s', %w", name, ErrExchangeNotFound)
+	}
+}
+
+// GetDefaultExchangeByName returns a defaulted exchange by its name if it exists.
+// This will allocate a new exchange and setup the default config for it. This
+// will automatically fetch available pairs.
+func GetDefaultExchangeByName(ctx context.Context, name string) (exchange.IBotExchange, error) {
```
Comment (gloriousCode) [inline engine/helpers.go:1061]: `CreateDefaultExchangeByName`/`NewExchangeByNameWithDefaults` or something since you're not really getting

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Thanks for addressing those (much) earlier nits! Just the most pedantic thing from me 💪

**Review by gloriousCode — APPROVED** [review]
tACK!

**Pull request merged**